### DISCUSSION TITLE
2022/08/16/mysteryofthefrogriddle

### DISCUSSION THREAD

**Discussion by giscus** [horenbergerb/horenbergerb.github.io#5] [Announcements]
# 2022/08/16/mysteryofthefrogriddle

Cite me in your thesis

https://horenbergerb.github.io/2022/08/16/mysteryofthefrogriddle.html

**Comment by horenbergerb** [comment]
Wow, I can add comments to my posts now?

**Comment by jjcrawford** [comment]
Sorry to necro, but it still keeps me up at night. I do believe ½ is a good answer but for fundamentally different reasons.

The whole sampling procedure of randomly generating samples from a broader space and then excluding them if they don't match the conditions you want is totally fine though — this is actually just how all Monte Carlo sampling works. So I do not agree with the reasoning of your "I did not select the clearing specifically to guarantee that it would have at least one male" statement.

But I think the "correct" way of arriving back at an answer of ½ is that if you interpret the frog croaking as itself being part of the random process (so if you allow e.g. every possible combination like "the first frog is male and croaked, the second frog is female and croaked" — 2⁴ such combinations) and then filter out the situations which are inconsistent with the observation that only one of the two frogs croaked and the one that croaked happened to be male, we magically arrive back at a situation where exactly ½ of the remaining states includes a female, matching our intuitions. You can quickly check this for yourself, but the remaining valid states we end up with are just:
- male croaked, male didn't croak
- male croaked, female didn't croak
- male didn't croak, male croaked
- female didn't croak, male croaked
of which obviously 2 of the 4 include a female.

It's still really quite a head-scratcher how interpreting the frog croak as just being a signal that "at least one is male" rather than considering the frog croak as an event in its own right could so substantially change the result, but the whole thing definitely highlights how Bayesian inference is full of crazy pitfalls.
